plugins/woocommerce-admin/client/marketplace/components/install-flow/install-new-product-modal.tsx

```diff
@@ -0,0 +1,401 @@
+				setNotice( {
+					status: 'error',
+					message: __(
+						'Activation failed. Please try again from the plugins page.',
+						'woocommerce'
+					),
+				} );
+			} );
+	}
+
+	function orderAndInstall() {
+		if ( ! product || ! product.id ) {
+			return;
+		}
+
+		recordEvent( 'marketplace_install_new_product_clicked', {
+			product_id: product.id,
+		} );
+
+		setInstallStatus( InstallFlowStatus.installing );
+
+		createOrder( product.id )
+			.then( ( response ) => {
+				// This narrows the CreateOrderResponse type to CreateOrderSuccessResponse
+				if ( ! response.success ) {
+					throw response;
+				}
+
+				const subscriptionData = response.data.subscription;
+
+				setSubscription( subscriptionData );
+
+				dispatch( installingStore ).startInstalling(
+					subscriptionData.product_key
+				);
+
+				return downloadProduct( subscriptionData ).then(
+					( downloadResponse ) => {
+						dispatch( installingStore ).stopInstalling(
+							subscriptionData.product_key
+						);
+
+						if ( downloadResponse.data.activateUrl ) {
+							setActivateUrl( downloadResponse.data.activateUrl );
+
+							setInstallStatus(
+								InstallFlowStatus.installedCanActivate
+							);
+						} else {
+							setInstallStatus(
+								InstallFlowStatus.installedCannotActivate
+							);
+						}
+					}
+				);
+			} )
+			.catch( ( error ) => {
+				/**
+				 * apiFetch doesn't return the error code in the error condition.
+				 * We'll rely on the data returned by the server.
+				 */
+				if ( error.data.redirect_location ) {
+					setNotice( {
+						status: 'warning',
+						message: __(
+							'Missing information. Redirecting to Woo.com to continue the installation.',
+							'woocommerce'
+						),
+					} );
+
+					window.location.href = error.data.redirect_location;
+				} else {
+					setInstallStatus( InstallFlowStatus.installFailed );
+					setNotice( {
+						status: 'error',
+						message:
+							error.data.message ??
+							__(
+								'An error ocurred. Please try again later.',
+								'woocommerce'
+							),
+					} );
+				}
+			} );
+	}
+
+	function onClose() {
+		navigateTo( {
+			url: getNewPath(
+				{
+					...query,
+					install: undefined,
+					installProduct: undefined,
+				},
+				MARKETPLACE_PATH,
+				{}
+			),
+		} );
+	}
+
+	function getTitle(): string {
+		if ( installStatus === InstallFlowStatus.activated ) {
+			return __( 'You are ready to go!', 'woocommerce' );
+		}
+
+		return __( 'Add to Store', 'woocommerce' );
+	}
+
+	function getDescription(): string {
+		if ( installStatus === InstallFlowStatus.notConnected ) {
+			return '';
+		}
+
+		if (
+			installStatus === InstallFlowStatus.installedCanActivate ||
+			installStatus === InstallFlowStatus.activating
+		) {
+			return __(
+				'Extension successfully installed. Would you like to activate it?',
+				'woocommerce'
+			);
+		}
+
+		if ( installStatus === InstallFlowStatus.installedCannotActivate ) {
```

Comment:
Sometimes WordPress doesn't allow us to activate the plugin, by omitting the `activateUrl`. It can either:
- Plugin is already active: this happens when plugin is deleted without deactivation and then re-installed. 
- User doesn't have the right capability. 

I guide user to the plugins page in those cases. If you look at the `primaryButton` you'll see the main action is **View Plugins**